test/cpp/c10d/TCPStoreTest.cpp

```diff
@@ -204,6 +205,50 @@ TEST(TCPStoreTest, testCleanShutdown) {
   clientThread.join();
 }
 
+TEST(TCPStoreTest, testLibUVPartialRead) {
+  int numWorkers = 2;
+
+  // server part
+  c10d::TCPStoreOptions server_opts{
+      0,
+      true, // is master
+      numWorkers,
+      false, // wait workers
+      std::chrono::seconds(defaultTimeout)};
+  server_opts.useLibUV = true;
+
+  auto serverTCPStore =
+      std::make_unique<c10d::TCPStore>("127.0.0.1", server_opts);
+
+  // client part
+  c10d::TCPStoreOptions client_opts{
+      serverTCPStore->getPort(),
+      false, // is master
+      numWorkers,
+      false, // wait workers
+      std::chrono::seconds(defaultTimeout)};
+  client_opts.useLibUV = true;
+  auto clientTCPStore =
+      c10::make_intrusive<c10d::TCPStore>("127.0.0.1", client_opts);
+  auto clientThread = std::thread([&clientTCPStore] {
+    std::string keyPrefix(
+        "/default_pg/0//b7dc24de75e482ba2ceb9f9ee20732c25c0166d8//cuda//");
+    std::string value("v");
+    std::vector<uint8_t> valueBuf(value.begin(), value.end());
+
+    // split store->set(key, valueBuf) into two requests
+    for (int i = 0; i < 10; ++i) {
+      std::string key = keyPrefix + std::to_string(i);
+      clientTCPStore->_splitSet(key, valueBuf);
+
+      // check the result on server
+      c10d::test::check(*clientTCPStore, key, "v");
+    }
+  });
+
```

Comment:
Do you also need `serverTCPStore = nullptr;` here?

Reply:
That will cause the issue. If we have it here, the server could be destructed before the client completes its execution.

Reply:
Then maybe add after clientThread Join?

Reply:
When the `unique_ptr` object `serverTCPStore` goes out of the scope (in this case it's the end of the test), the object will be destroyed (same as `serverTCPStore = nullptr`) and the TCPStore will be freed. So adding `serverTCPStore = nullptr` will just be the same as the current code.

Reply:
I guess my point is to set to null after join, by the time of join, all clients should have finished? But since this is a unique pointer, it should be fine.